DataFormats/Provenance/src/IndexIntoFile.cc

```diff
@@ -1815,32 +1847,40 @@ namespace edm {
     if (type() == kLumi) {
       auto entry = runOrLumisEntry(indexToLumi()).entry();
       if (entry == invalidEntry) {
-        if (indexToLumi() + 1 < size()) {
-          if (runOrLumisEntry(indexToLumi()).lumi() != runOrLumisEntry(indexToLumi() + 1).lumi()) {
-            //find the end of this lumi
-            auto const& runLumiEntry = runOrLumisEntry(indexToLumi());
-            for (auto nextIndex = indexToLumi() + 1; nextIndex < size(); ++nextIndex) {
-              auto const& nextRunLumiEntry = runOrLumisEntry(nextIndex);
-              if (runLumiEntry.lumi() == nextRunLumiEntry.lumi() and runLumiEntry.run() == nextRunLumiEntry.run() and
-                  runLumiEntry.processHistoryIDIndex() == nextRunLumiEntry.processHistoryIDIndex()) {
-                auto nextEntry = nextRunLumiEntry.entry();
-                if (nextEntry != invalidEntry) {
-                  return nextEntry;
-                }
-              }
-            }
-            return continuedLumi;
+        auto const& runLumiEntry = runOrLumisEntry(indexToLumi());
+        for (int index = indexToLumi() + 1; index < indexedSize_; ++index) {
+          auto const& laterRunOrLumiEntry = runOrLumisEntry(index);
+          if (runLumiEntry.lumi() == laterRunOrLumiEntry.lumi() and runLumiEntry.run() == laterRunOrLumiEntry.run() and
```

Comment:
It is very close to the same, but there are some differences.

The new version throws if it does not find a valid entry number, although that should never happen. The old version sometimes returned invalidEntry and sometimes continuedLumi.

I could not see that the continuedLumi constant was ever used anywhere, so I deleted it here and where the constant was defined.

More significant, it is possible to have a sequence like invalid-valid-invalid-valid, all with the same lumi number. The old version could not handle the second invalid in that sequence and would return invalidEntry. This can occur with multiple non-contiguous sequences of events from the same run in noRunLumiSort mode. (The second removed line was causing this problem).